### PR TITLE
M2-5915: [Admin Panel] - Item Flow Conditional Logic - Add validation if Item from the Summary row is already in use in another Conditional card

### PR DESCRIPTION
**Pull request by alex-i-e**
- [x] Tests for the changes have been added
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-5915](https://mindlogger.atlassian.net/browse/M2-5915)

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| image | image |

### CODE REVIEW

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import { useCurrentActivity } from 'modules/Builder/hooks';
+import { ConditionalLogic } from 'shared/state/Applet';
+
+import { getItemsInUsage } from './SummaryRow.utils';
+
+export const useItemsInUsage = (name: string) => {
+  const { fieldName } = useCurrentActivity();
+  const itemKey = useWatch({ name: `${name}.itemKey` });
+  const conditionalLogic: ConditionalLogic[] = useWatch({ name: `${fieldName}.conditionalLogic` });
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts]: I think these two useWatch calls can be merged into one.
Something like:
const [itemKey, conditionalLogic]: [types] = useWatch({ name: ['name1', 'name2'});

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts:1]: Should we consider adding test coverage for this hook, or is the coverage for the utils sufficient?

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        itemKey: undefined,
+      },
+    ];
+
+    test.each`
+      conditionalLogic    | itemKey      | expectedArrayinSet      | description
+      ${conditionalLogic} | ${'item-2'}  | ${['item-1', 'item-3']} | ${'returns items when current is chosen'}
+      ${conditionalLogic} | ${undefined} | ${['item-1', 'item-3']} | ${'returns items when current is empty'}
+      ${conditionalLogic} | ${'item-1'}  | ${['item-3']}           | ${'returns filtered items when overlapped'}
+    `('$description', async ({ conditionalLogic, itemKey, expectedArrayinSet }) => {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.test.ts]:
```suggestion
    `('$description', async ({ conditionalLogic, itemKey, expectedArrayInSet }) => {
```

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.ts [hunk 3]
```diff
@@ -18,10 +19,34 @@ export const getMatchOptions = () => [
   },
 ];
 
-export const getItemsOptions = (items: ItemFormValues[]) =>
+export const getItemsOptions = ({
+  items,
+  itemsInUsage,
+}: {
+  items: ItemFormValues[];
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.ts]: Could you please move types to SummaryRow.types.ts?

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.ts [hunk 4]
```diff
@@ -18,10 +19,28 @@ export const getMatchOptions = () => [
...
+      return [...optionList, { value, labelKey: item.name, disabled, tooltip }];
+    }
 
     return optionList;
   }, []);
+
+export const getItemsInUsage = ({
+  conditionalLogic,
+  itemKey,
+}: {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.utils.ts]: Also move this type to types, please.

File: src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts
Comment (alex-i-e) [reply src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.hooks.ts:1]: yeap, mainly, it's covered in utils

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**